app/Http/Controllers/SignupsController.php

```diff
@@ -124,6 +124,17 @@ public function index(Request $request)
      */
     public function show(Request $request, Signup $signup)
     {
+        // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
+        if (starts_with($request->query('include'), 'posts')) {
+            $types = (new \League\Fractal\Manager)
+                ->parseIncludes($request->query('include'))
+                ->getIncludeParams('posts');
+
+            $types = $types ? $types->get('type') : null;
+
+            $signup = $signup->withVisiblePosts($types)->get()->keyBy('id')[$signup->id];
```

Comment:
This seems suspect! It looks like we're loading all signups into memory, running a transformation on that array, and then grabbing an index. This gets our test passing, but would probably choke in production (we have a lot of signups!!)

Reply:
How about something like this, where we load the relationship onto the model (using `load`) and then customize it w/ a closure to limit it by the `type` column (similar to how we handled eager-loading before):

```diff
             $types = $types ? $types->get('type') : null;

-            $signup = $signup->withVisiblePosts($types)->get()->keyBy('id')[$signup->id];
+            $signup->load(['visiblePosts' => function ($query) use ($types) {
+                if ($types) {
+                    $query->whereIn('type', $types);
+                }
+            }]);
         }

         return $this->item($signup, 200, [], $this->transformer, $request->query('include'));
```

Reply:
ah that is so true - I need to keep that top of mind when using queries and large datasets for performance! Updated with what you said - thanks @DFurnes !